### ISSUE TITLE
NPE with the new released compiler

### ISSUE THREAD

**Issue by arichiardi** [thheller/shadow-cljs#401]
Hi Thomas, we are at the bleeding edge here so I tried the new `1.10.439` and I got this back:

```
------ ERROR -------------------------------------------------------------------
 File: jar:file:/home/arichiardi/.m2/repository/org/clojure/clojurescript/1.10.439/clojurescript-1.10.439.jar!/cljs/core.cljs
NullPointerException: 
	cljs.compiler/emit/fn--3144 (compiler.cljc:189)
	clojure.lang.Atom.swap (Atom.java:37)
	clojure.core/swap! (core.clj:2344)
	clojure.core/swap! (core.clj:2337)
	cljs.compiler/emit (compiler.cljc:187)
	cljs.compiler/emit (compiler.cljc:182)
	shadow.build.compiler/eval12921/fn--12922 (compiler.clj:232)
	clojure.lang.MultiFn.invoke (MultiFn.java:234)
	shadow.build.compiler/do-compile-cljs-resource/fn--12984/fn--12987 (compiler.clj:454)
	shadow.build.compiler/do-compile-cljs-resource/fn--12984 (compiler.clj:451)
	shadow.build.compiler/do-compile-cljs-resource (compiler.clj:414)
	shadow.build.compiler/do-compile-cljs-resource (compiler.clj:376)
	shadow.build.compiler/maybe-compile-cljs/fn--13070 (compiler.clj:694)
	shadow.build.compiler/maybe-compile-cljs (compiler.clj:693)
	shadow.build.compiler/maybe-compile-cljs (compiler.clj:669)
	shadow.build.compiler/par-compile-one (compiler.clj:795)
	shadow.build.compiler/par-compile-one (compiler.clj:750)
	shadow.build.compiler/par-compile-cljs-sources/fn--13104/iter--13126--13130/fn--13131/fn--13132/fn--13133 (compiler.clj:868)
	clojure.core/apply (core.clj:657)
	clojure.core/with-bindings* (core.clj:1965)
	clojure.core/with-bindings* (core.clj:1965)
	clojure.core/apply (core.clj:661)
	clojure.core/bound-fn*/fn--5649 (core.clj:1995)
	java.util.concurrent.FutureTask.run (FutureTask.java:266)
	java.util.concurrent.ThreadPoolExecutor.runWorker (ThreadPoolExecutor.java:1149)
	java.util.concurrent.ThreadPoolExecutor$Worker.run (ThreadPoolExecutor.java:624)
	java.lang.Thread.run (Thread.java:748)
```

Thanks for your wonderful work on shadow!

**Comment by thheller** [comment]
`cljs.compiler/*source-map-data-gen-col*` was introduced as a new binding.

Additionally some of the AST data was changed. I already fixed the issues in a branch but want to do some more testing first.

**Comment by thheller** [comment]
Found a regression in 1.10.439. Probably holding off on the update as this seems to make inference fairly unreliable in a few cases. A build that didn't produce any inference warnings on 1.10.339 now produces 10 and all of them incorrect.

https://dev.clojure.org/jira/browse/CLJS-2957

**Comment by arichiardi** [comment]
Oh the ast change is not good news for tooling - and the regression too. Thank you for the update.

**Comment by thheller** [comment]
This is now fixed in `2.7.0`.
